core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -242,13 +241,14 @@ class SocketServer(val config: KafkaConfig,
   }
 
   private def endpoints = config.listeners.map(l => l.listenerName -> l).toMap
+  private def endpointLogContext(endpoint: EndPoint): LogContext = new LogContext(s"[$logPrefix, listener=${endpoint.listenerName.value}] ")
 
   protected def createDataPlaneAcceptor(endPoint: EndPoint, isPrivilegedListener: Boolean, requestChannel: RequestChannel): DataPlaneAcceptor = {
-    new DataPlaneAcceptor(this, endPoint, config, nodeId, connectionQuotas, time, isPrivilegedListener, requestChannel, metrics, credentialProvider, logContext, memoryPool, apiVersionManager)
+    new DataPlaneAcceptor(this, endPoint, config, nodeId, connectionQuotas, time, isPrivilegedListener, requestChannel, metrics, credentialProvider, endpointLogContext(endPoint), memoryPool, apiVersionManager)
```

Comment:
We have 2 abstract methods in Acceptor:
```
  def metricPrefix(): String
  def threadPrefix(): String
```
how about adding another similar method: `def logPrefix(): String`

Reply:
It's not immediately obvious to me how this would work.
We still need to pass down some form of log prefix from the SocketServer to the acceptor, assuming we want similar formatting between SocketServer logs and Acceptor/Processor logging.